app/views/articles/_sticky_nav.html.erb

```diff
@@ -58,6 +58,7 @@
 <%# cache("article-sidebar-content-#{rand(5)}-#{@article.id}-#{user_signed_in?}-#{(@organization || @user).latest_article_updated_at}", expires_in: 15.minutes) do %>
   <% sidebar_ad = DisplayAd.for_display(area: "post_sidebar",
                                         user_signed_in: user_signed_in?,
+                                        user_id: current_user&.id,
```

Comment:
`current_user` in these views is currently not possible, because of edge caching, effectively meaning that this PR depends on #19227 before it can go further.